sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -160,7 +166,248 @@ def plot(self, *, include_values=True, cmap='viridis',
+            display_labels=display_labels,
+            include_values=include_values,
+            cmap=cmap,
+            ax=ax,
+            xticks_rotation=xticks_rotation,
+            values_format=values_format,
+            colorbar=colorbar,
+        )
+
+    @classmethod
+    @inject_docstring(common_docstring=_common_docstring_from_methods)
+    def from_predictions(
+        cls,
+        y_true,
+        y_pred,
+        *,
+        labels=None,
+        sample_weight=None,
+        normalize=None,
+        display_labels=None,
+        include_values=True,
+        xticks_rotation="horizontal",
+        values_format=None,
+        cmap="viridis",
+        ax=None,
+        colorbar=True,
+    ):
+        """Plot Confusion Matrix given true and predicted labels.
+
+        Read more in the :ref:`User Guide <confusion_matrix>`.
+
+        .. versionadded:: 0.24
+
+        Parameters
+        ----------
+        y_true : array-like of shape (n_samples,)
+            True labels.
+
+        y_pred : array-like of shape (n_samples,)
+            The predicted labels given by the method `predict` of an
+            classifier.
+
+        {common_docstring}
+
+        See Also
+        --------
+        ConfusionMatrixDisplay.from_estimator : Plot the confusion matrix
+            given an estimator, the data, and the label.
+
+        Examples
+        --------
+        >>> import matplotlib.pyplot as plt  # doctest: +SKIP
+        >>> from sklearn.datasets import make_classification
+        >>> from sklearn.metrics import ConfusionMatrixDisplay
+        >>> from sklearn.model_selection import train_test_split
+        >>> from sklearn.svm import SVC
+        >>> X, y = make_classification(random_state=0)
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...         X, y, random_state=0)
+        >>> clf = SVC(random_state=0)
+        >>> clf.fit(X_train, y_train)
+        SVC(random_state=0)
+        >>> y_pred = clf.predict(X_test)
+        >>> ConfusionMatrixDisplay.from_predictions(
+        ...    y_test, y_pred)  # doctest: +SKIP
+        >>> plt.show()  # doctest: +SKIP
+        """
+        check_matplotlib_support(f"{cls.__name__}.from_predictions")
+
+        y_true_type = type_of_target(y_true)
+        y_pred_type = type_of_target(y_pred)
+        supported_type = {"binary", "multiclass"}
+        if y_true_type not in supported_type:
+            raise ValueError(
+                f"Only 'binary' and 'multiclass' classification problems are "
+                f"supported. Got {y_true_type}."
+            )
+        elif y_pred_type not in supported_type:
+            raise ValueError(
+                f"The type of target of `y_pred` differs from the one of "
+                f"`y_true`. Ensure to pass prediction as given by "
+                f"`classifier.predict(X)`. Got `y_pred` of type "
+                f"'{y_pred_type}'."
+            )
+        check_consistent_length(y_true, y_pred)
```

Comment:
We would not need to validate for `from_predictions` because validation is handled by `confusion_matrix`?

Reply:
Yep, the message is slightly more cryptic but understandable.